### PR TITLE
Refactor your code

### PR DESCRIPTION
**Pull request by AshwinCarvalho999**
### Learning objectives

- Described the SOLID principles of OOP.
- Refactored our own Ruby code after gaining new knowledge.
- Used YAGNI, KISS, and DRY rules and RUBY language style guides in code.
- Follow SOLID principles.
- Payed special attention to the "Single-responsibility Principle".

### CODE REVIEW

File: choose_action.rb [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-end
+# def choose_action(selected_option)
+#   case selected_option
+#   when 1
+#     display_all_people
+#     run
+#   when 2
+#     display_all_books
+#     run
+#   when 3
```
Comment (Davidosky007) [inline choose_action.rb:9]: * Kindly fix this error I get when I try to create a person, you are importing a method (choose action method) that  has been commented out that's why it's throwing the error, Please fix it 🙏 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @AshwinCarvalho999 @j-tee 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @AshwinCarvalho999 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mckent05 — COMMENTED** [review]
Hi @AshwinCarvalho999, 

# STATUS: INVALID REQUEST ❌ ❌ 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

 There is currently no way for me to run your project, I cannot find the `main.rb` file. You seem to have not merged your previous work into the development branch. Please merge the changes from the previous milestone and pull the changes in your current branch. Please fix this 🙏 

Your repo:
fwfwf

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**